molgenis-api/src/main/java/org/molgenis/api/model/response/PageResponse.java

```diff
@@ -0,0 +1,22 @@
+package org.molgenis.api.model.response;
+
+import com.google.auto.value.AutoValue;
+import org.molgenis.util.AutoGson;
+
+@AutoValue
+@AutoGson(autoValueClass = AutoValue_PageResponse.class)
+@SuppressWarnings(
+    "squid:S1610") // Abstract classes without fields should be converted to interfaces
+public abstract class PageResponse {
+  public abstract int getSize();
+
+  public abstract int getTotalElements();
+
+  public abstract int getTotalPages();
+
+  public abstract int getNumber();
+
+  public static PageResponse create(int size, int totalElements, int totalPages, int number) {
```

Comment:
Validate that inputs are positive: https://github.com/google/auto/blob/master/value/userguide/builders-howto.md#-validate-property-values